### PR TITLE
Add docs for More debugging options in DevTools

### LINKED ISSUE #6750: Add debugging options docs
### What information is missing?

Add docs for debugging options to the [performance view page](https://docs.flutter.dev/development/tools/devtools/performance).

Screen Shot 2022-01-28 at 3 09 13 PM

### How would you like us to fix this problem?

_No response_

### PR DESCRIPTION
**Pull request by johnpryan**
This adds documentation for the "More debugging options" in Devtools.

cc: @kenzieschmoll 

fixes #6750

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/development/tools/devtools/performance.md [hunk 1]
```diff
@@ -153,8 +153,30 @@ enable the Track Layouts option:
 To see render object paint events in the timeline,
 enable the Track Paints option:
 
+
 ![Screenshot of track paints]({{site.url}}/assets/images/docs/tools/devtools/track-paints.png)
 
+## More debugging options
```
Comment (kenzieschmoll) [inline src/development/tools/devtools/performance.md:159]: can you add a note that users will have to reproduce activity in their app to see differences in performance

Comment (johnpryan) [reply src/development/tools/devtools/performance.md:159]: Added a couple of paragraphs

File: src/development/tools/devtools/performance.md [hunk 2]
```diff
@@ -158,6 +158,16 @@ enable the Track Paints option:
...
 
+To diagnose performance problems related to rendering layers,
+toggle off a rendering layer.
+These options are enabled by default.
+
+To see the effects on your app's performance,
+reproduce the activity in your app,
+select the new frames in the frames chart,
+to inspect the timeline events
+with the layers disabled.
```
Comment (kenzieschmoll) [inline src/development/tools/devtools/performance.md:169]: maybe add a sentence here that says. "If Paint time or Raster time has significantly decreased, excessive use of the effects you disabled may be contributing to the jank you saw in your app."

@dnfield can you check me here - if a user disables clip layers and sees a performance improvement, where is that improvement likely to show up? In the Paint event on the UI thread? In the raster events?

Comment (dnfield) [reply src/development/tools/devtools/performance.md:169]: Raster.

Comment (kenzieschmoll) [reply src/development/tools/devtools/performance.md:169]: thanks @dnfield . and is this true for all the layer options here? (clip, opacity, physical shape)

Comment (dnfield) [reply src/development/tools/devtools/performance.md:169]: In general, yes - although _some_ build time improvements on the UI thread may be noted, these are things that have more impact on raster time.

Comment (dnfield) [reply src/development/tools/devtools/performance.md:169]: (i.e. clipping (especially clipping with anti-aliasing if it is requested), saving layers, drawing shadows)

Comment (johnpryan) [reply src/development/tools/devtools/performance.md:169]: added a sentence for this in [fe1a818](https://github.com/flutter/website/pull/6975/commits/fe1a818936387c4d33d09b18f3f6fc95edfa9563)

**Review by domesticmouse — APPROVED** [review]

File: src/development/tools/devtools/performance.md [hunk 3]
```diff
@@ -153,8 +153,43 @@ enable the Track Layouts option:
...
+If Paint time or Raster time has significantly decreased,
```
Comment (kenzieschmoll) [inline src/development/tools/devtools/performance.md:170]: based on Dan's response above, it sounds like this should be just "If Raster time..." and not Paint

**Review by sfshaza2 — APPROVED** [review]
It sounds like one more small tweak is needed (paint or raster => raster), then LGTM

File: src/development/tools/devtools/performance.md [hunk 4]
```diff
@@ -153,8 +153,43 @@ enable the Track Layouts option:
...
+
+To diagnose performance problems related to rendering layers,
+toggle off a rendering layer.
+These options are enabled by default.
+
+To see the effects on your app's performance,
+reproduce the activity in your app,
+select the new frames in the frames chart,
+to inspect the timeline events
+with the layers disabled.
```
Comment (sfshaza2) [inline src/development/tools/devtools/performance.md:169]:
```suggestion
To see the effects on your app's performance,
reproduce the activity in your app.
Then select the new frames in the frames chart
to inspect the timeline events
with the layers disabled.
```

File: src/development/tools/devtools/performance.md [hunk 5]
```diff
@@ -153,8 +153,43 @@ enable the Track Layouts option:
...
+If Raster time has significantly decreased,
+excessive use of the effects you disabled may be contributing
```
Comment (sfshaza2) [inline src/development/tools/devtools/performance.md:171]:
```suggestion
excessive use of the effects you disabled might be contributing
```

File: src/development/tools/devtools/performance.md [hunk 6]
```diff
@@ -153,8 +153,43 @@ enable the Track Layouts option:
...
+  to check whether excessive use of opacity effects
+  are affecting performance.
+  If performance improves with this option disabled,
+  try to reduce the use of opacity effects in your app.
+- **Render Physical Shape Layers:** Disable this option
+  to check whether excessive use of physical modeling effects
+  are affecting performance,
+  such as shadows or elevation.
+  If performance improves with this option disabled,
+  try to reduce the use of physical modeling effects in your app.
```
Comment (sfshaza2) [inline src/development/tools/devtools/performance.md:189]:
```suggestion
**Render Clip layers**
:  Disable this option  to check whether excessive use of clipping
  is affecting performance.
  If performance improves with this option disabled,
  try to reduce the use of clipping effects in your app.
  
**Render Opacity layers**
:  Disable this option to check whether
   excessive use of opacity effects are affecting performance.
   If performance improves with this option disabled,
   try to reduce the use of opacity effects in your app.
  
**Render Physical Shape Layers**
:  Disable this option to check whether excessive
   use of physical modeling effects are affecting performance,
  such as shadows or elevation.
  If performance improves with this option disabled,
  try to reduce the use of physical modeling effects in your app.
```

**Review by sfshaza2 — APPROVED** [review]
A few little format/word things, but otherwise LGTM

**Pull request merged**